client/src/components/Form/Elements/FormFTP.vue

```diff
@@ -0,0 +1,193 @@
+<template>
+    <div :class="model.cls">
+        <div v-show="waiting" class="upload-ftp-wait fa fa-spinner fa-spin" />
+        <div v-show="model.help_enabled" class="upload-ftp-help">{{ model.help_text }}</div>
+        <div v-show="!waiting" class="upload-ftp-content">
+            <span style="whitespace: nowrap; float: left">Available files: </span>
+            <span style="whitespace: nowrap; float: right">
+                <span class="upload-icon fa fa-file-text-o" />
+                <span class="upload-ftp-number">{{ ftpFiles.length }} files</span>
+                <span class="upload-icon fa fa-hdd-o" />
+                <span class="upload-ftp-disk">{{ bytesToString(totalSize, true) }}</span>
+            </span>
+            <table class="grid" style="float: left">
+                <thead>
+                    <tr>
+                        <th v-show="model.collection" class="_has_collection">
+                            <div
+                                class="upload-ftp-select-all"
+                                :class="{ [model.class_add]: !isAllSelected, [model.class_remove]: isAllSelected }"
+                                @click="selectAll"></div>
+                        </th>
+                        <th>Name</th>
+                        <th>Size</th>
+                        <th>Created</th>
+                    </tr>
+                </thead>
+                <tbody class="upload-ftp-body">
+                    <tr
+                        v-for="(ftpFile, index) in ftpFiles"
+                        :key="ftpFile.path"
+                        class="upload-ftp-row"
+                        @click="onRowClick(ftpFile)">
+                        <td class="ftp-name" colspan="4">
+                            <div
+                                class="icon"
+                                :class="{
+                                    [model.class_add]: !isSelected(ftpFile),
+                                    [model.class_remove]: isSelected(ftpFile),
+                                }"></div>
+                            {{ ftpFile.path }}
+                        </td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+        <div v-show="ftpFiles.length === 0" class="upload-ftp-warning warningmessage">no files.</div>
+    </div>
+</template>
+
+<script>
+import _ from "underscore";
+import $ from "jquery";
```

Comment:
Please avoid using both jquery and underscore; we should not need either of these in our components.

Reply:
Sure thing